### DISCUSSION TITLE
modify k3s --etcd-snapshot-dir Backup path prompt no such file or directory

### DISCUSSION THREAD

**Discussion by susugo** [k3s-io/k3s#8046] [Q&A] [answered]
My configuration is as follows:
        '--etcd-snapshot-dir' \
        '/data/lib/rancher/k3s/server/db/snapshots' \
Log error, do I need to manually create my configured path every time、How to make him automatically create this directory
```
Jul 25 10:14:36 jsdesign k3s[26265]: time="2023-07-25T10:14:36.169714159+08:00" level=error msg="Failed to record snapshots for cluster: open /data/lib/rancher/k3s/server/db/snapshots: no such file or directory"
```

**Comment by brandond** [comment, marked as answer]
If you set `--data-dir=/data/lib/rancher/k3s`, the snapshot dir would be relocated alongside the rest of the K3s data, and automatically created as well.

If you explicitly set the snapshot-dir to an unmanaged path, you would need to ensure that it actually exists, yes.
